frontend/src/metabase/core/components/Link/Link.tsx

```diff
@@ -19,14 +19,15 @@ const Link = ({
   children,
   disabled,
   tooltip,
+  tabIndex,
   ...props
 }: LinkProps): JSX.Element => {
   const link = (
     <LinkRoot
       {...props}
       to={to}
       disabled={disabled}
-      tabIndex={disabled ? -1 : undefined}
+      tabIndex={tabIndex ?? disabled ? -1 : undefined}
```

Comment:
Allow overriding the `tabIndex`. This is used in https://github.com/metabase/metabase/pull/23833/files#diff-eeeb01acb404315c0c6c49673896cc649a1339bd90341aaced2e75eb9c9d4af0R19 to prevent <kbd>Tab</kbd> selection.

Reply:
This expression feels a bit confusing to me. My first thought was that `tabIndex ?? disabled` is the ternary condition. However, it's actually `tabIndex` OR `disabled ? -1 : undefined`, right? Should we maybe extract this block? Something like `useMemo` can make it a bit more straightforward I believe:

```js
const tabIndex = useMemo(() => {
  if (tabIndex) {
     return tabIndex;
  }
  return disabled ? -1 : undefined;
}, [ ... ])
```

Reply:
Yeah, you're right. It might be obvious to me now, but this could confuse other people. +1 for simplifying this.